src/Castle.Windsor/MicroKernel/IHandler.cs

```diff
@@ -82,5 +82,11 @@ public interface IHandler : ISubDependencyResolver
 		/// </summary>
 		/// <returns></returns>
 		object TryResolve(CreationContext context);
+
+		/// <summary>
+		/// Needed to support root scope for multiple container when integrating with .NET 8
+		/// DI engine.
+		/// </summary>
+		IKernel GetKernel();
```

Comment:
This is a fairly severe breaking change to `IHandler`, need to think about this more.